scene/2d/tile_map_layer.cpp

```diff
@@ -932,7 +934,18 @@ void TileMapLayer::_physics_draw_cell_debug(const RID &p_canvas_item, const Vect
 				const RID &shape = ps->body_get_shape(body, shape_index);
 				const PhysicsServer2D::ShapeType &type = ps->shape_get_type(shape);
 				if (type == PhysicsServer2D::SHAPE_CONVEX_POLYGON) {
-					rs->canvas_item_add_polygon(p_canvas_item, ps->shape_get_data(shape), color);
+					Vector<Vector2> debug_polygon_vertices = ps->shape_get_data(shape);
+
+					Vector<Color> debug_face_colors;
```

Comment:
This was outside the loop to avoid creating a new vector at every iteration. Please keep it that way for performance reasons.

Reply:
Moved the Vector<Color> for the face and edge colors outside the loop.